lib/applications/resolvers.ts

```diff
@@ -6,14 +6,109 @@ import {
   ApplicationFieldTooLongError,
 } from '@lib/applications/errors'; // Application errors
 import { DBErrorCode } from '@lib/db/errors'; // Database errors
+import { SortOrder } from '@tools/types'; // Sorting type
 
 /**
  * Query all the RCD applications in the internal-facing app
  * @returns All RCD applications
  */
-export const applications: Resolver = async (_parent, _args, { prisma }) => {
-  const applications = await prisma.application.findMany();
-  return applications;
+export const applications: Resolver = async (_parent, { filter }, { prisma }) => {
+  const { order, permitType, requestType, status, search, limit = 20, offset = 0 } = filter;
+
+  // Parse search string
+  let userIDSearch, firstSearch, middleSearch, lastSearch;
+
+  if (parseInt(search)) {
+    userIDSearch = parseInt(search);
+  } else if (search) {
+    [firstSearch, middleSearch, lastSearch] = search.split(' ');
+    middleSearch = middleSearch || firstSearch;
+    lastSearch = lastSearch || middleSearch;
+  }
+
+  // Parse sorting order
+  let orderBy = undefined;
+
+  if (order && order.length > 0) {
+    const sortingOrder: Array<Record<string, SortOrder>> = [];
+    order.forEach(([field, order]: [string, SortOrder]) => {
+      if (field === 'name') {
+        // Primary sort is by first name and secondary sort is by last name
+        sortingOrder.push({ firstName: order });
+        sortingOrder.push({ lastName: order });
+      } else if (field === 'dateReceived') {
+        sortingOrder.push({ createdAt: order });
+      }
+    });
+    orderBy = sortingOrder;
+  }
+
+  // Get number of applications with desired filters
+  const applicationsCount = await prisma.application.count({
+    where: {
+      applicant: {
+        id: userIDSearch,
+      },
+      applicationProcessing: {
+        status: status || undefined,
+      },
+      isRenewal: requestType ? requestType === 'Renewal' : undefined,
+      permitType: permitType || undefined,
+      AND: [
+        {
+          OR: [
+            { firstName: { contains: firstSearch, mode: 'insensitive' } },
+            { middleName: { contains: middleSearch, mode: 'insensitive' } },
+            { lastName: { contains: lastSearch, mode: 'insensitive' } },
+          ],
+        },
+      ],
+    },
```

Comment:
So you don't have to repeat this in both the count query and actual query you can take out the `where` in a variable and pass it to both queries. it will also be easier to make a change in one place in the future if needed